### DISCUSSION TITLE
Implementation of More Languages

### DISCUSSION THREAD

**Discussion by zanderlewis** [elide-dev/elide#921] [Ideas]
# Things that could be added

## Things to be addressed:
- GraalLua is 3y out of date, not complete. Action: Continue to make it on our own.
- Creation of a GraalPHP (Out of date by 4y). Action: Create an up-to-date GraalPHP.

## Possible Additions:
- [FastR](https://github.com/oracle/fastr) (GraalVM R implementation)

**Comment by sgammon** [comment]
Smalltalk (would be kinda exotic):
https://github.com/hpi-swa/trufflesqueak

CUDA integration:
https://github.com/NVIDIA/grcuda

Pkl is already on the way:
https://github.com/apple/pkl

.NET has some experiments:
https://github.com/jagotu/BACIL

**Reply by zanderlewis** [reply]
We would need to make an up-to-date version of CUDA. Last commit was 4y ago.

**Reply by sgammon** [reply]
GraalVM has been moving very quickly. Pkl was out of date by a few versions, but [upgrading it wasn't so hard](https://github.com/apple/pkl/pull/204) in the end. I should try the CUDA one out and see how bad it is.

**Reply by zanderlewis** [reply]
Made a [repo](https://github.com/WolfTheDeveloper/graal) of everything we need compiled together (and other things people might like). Hope you like it.
